---
title: "Introduction"
author: Charlie PAUVERT
classoption: a4paper
fontsize: 12pt
lang: en
output:
  pdf_document:
    pandoc_args: [ "--latex-engine=/usr/bin/pdflatex" ] 
    toc: yes
    number_section: true
header-includes:
 - \usepackage{libertine}
--- 

# Introduction (7-15p)


Outline : around a metagenomics analysis tool

## Food-Microbiome projects

We spend a fair amount of time eating, or thinking about it. 
Food is paradoxically both essential and trivial.
Multiple factors drives food-related studies: from scientific curiosity to yield and quality improvements.
Fermented foods - like cheese, sausage and beer - provides a level of diversity and complexity worth studying given their world-wide spread.

### Why? (1p)

Industrial partners and academics both wants insights into cheese ecosystems.
Their common interests were crystallized into two projects:

* Food-Microbiome project
* Food-Microbiome Transfert project

#### Food-Microbiome project

The Food-Microbiome project aims to deeply understand cheese ecosystems given next-generation sequencing technologies potential.
It is coordinated by Pierre RENAULT (PhD) and funded by the ANR -_The French National Research Agency_-. It intend to provide a proof-of-concept and preliminary results concerning the use of metagenomics for cheese ecosystems exploration.

#### Food-Microbiome Transfert project

##### Distribution of a metagenomics analysis tool

* Knowledge acquired during the previous project will be transfer to industrial partners and scientific collaborators.
* A comprehensive database and a website is developed to this end.
* It will be possible to analyse metagenomics data with our tool directly from the interface.

##### Experiments

* Aim: Unravel technological abilities/potential of unknown strains.
* Annotation improvements through wet lab experiments.


### Who? (0.75-1p)

#### Personal involvement in both projects

* Mainly development and tests in FM.
* Knowledge/know-how transfer and integration in FMT.

#### Several actors and contributors to these projects

* Academics: INRA mainly (Jouy, Grignon and Aurillac)
* Industrials partners and CNIEL -_French Dairy Interbranch Organization_ -: define needs.

* BAC Team:
	+ PR: project leader.
	+ ALA, CP, TG: development bioinformatics team.
	+ BD: functional strains characterization. Wet lab.
* MaIage unit :
	+ VL, SD, (TG): infrastructure team, platform administrator, technical advices
	+ MM, SB: statistical expertise.

* Figure from CNIEL presentation to summarise how data flows and by who is it produced or handled.

## Cheese ecosystems particularities (1p)

* Define micro-organisms and taxonomy hierarchy.
* Ecosystems described as a _good_ model : both simple and complex (see Dutton)


### Why simple ?

* Numbers of species compared to others.
* Milk is an accessible media for culturing.
* Wide overall knowledge : literature available

### Why complex ?

* Sometimes (everytime) cheese is _not only_ milk and micro-organisms
* Environment influence is significant (others sources micro-organisms, possible interactions, abiotic parameters) (see Irlinger)

## State-of-art tools for ecosystem exploration (2.5-3p)

### Methods (1p)

* Focus on metagenomics.
* Different sequencing approaches and their limits.
* See Ercolini 2013 for next-generation sequencing and food microbial ecology

### Post sequencing strategies (1.5-2p)

* Presentation of different types of metagenomics data handling strategies.
* Different tools and their limits in ecosystem exploration.

## Aims

* Summary of a two-year apprenticeship.
* Exploration of strategy taking into account potential divergence between environmental and reference genomes.
