---
title: "Introduction"
author: Charlie PAUVERT
classoption: a4paper
fontsize: 12pt
lang: en
output:
  pdf_document:
    pandoc_args: [ "--latex-engine=/usr/bin/pdflatex" ] 
    toc: yes
    number_section: true
header-includes:
 - \usepackage{libertine}
--- 

# Introduction (7-15p)


Outline : around a metagenomics analysis tool

## Food-Microbiome projects

We spend a fair amount of time eating, or thinking about it. 
Food is paradoxically both essential and trivial.
Multiple factors drives food-related studies: from scientific curiosity to yield and quality improvements.
Fermented foods - like cheese, sausage and beer - provides a level of diversity and complexity worth studying given their world-wide spread.

### Why? (1p)

Industrial partners and academics both wants insights into cheese ecosystems.
Their common interests were crystallized into two projects:

* Food-Microbiome project
* Food-Microbiome Transfert project

#### Food-Microbiome project

The Food-Microbiome project aims to deeply understand cheese ecosystems given next-generation sequencing technologies potential.
It is coordinated by Pierre RENAULT (PhD) and funded by the ANR -_The French National Research Agency_-. It intend to provide a proof-of-concept and preliminary results concerning the use of metagenomics for cheese ecosystems exploration.

#### Food-Microbiome Transfert project

##### Distribution of a metagenomics analysis tool

* Knowledge acquired during the previous project will be transfer to industrial partners and scientific collaborators.
* A comprehensive database and a website is developed to this end.
* It will be possible to analyse metagenomics data with our tool directly from the interface.

##### Experiments

* Aim: Unravel technological abilities/potential of unknown strains.
* Annotation improvements through wet lab experiments.


### Who? (0.75-1p)

#### Personal involvement in both projects

* Mainly development and tests in FM.
* Knowledge/know-how transfer and integration in FMT.

#### Several actors and contributors to these projects

* Academics: INRA mainly (Jouy, Grignon and Aurillac)
* Industrials partners and CNIEL -_French Dairy Interbranch Organization_ -: define needs.

* BAC Team:
	+ PR: project leader.
	+ ALA, CP, TG: development bioinformatics team.
	+ BD: functional strains characterization. Wet lab.
* MaIage unit :
	+ VL, SD, (TG): infrastructure team, platform administrator, technical advices
	+ MM, SB: statistical expertise.

* Figure from CNIEL presentation to summarise how data flows and by who is it produced or handled.

Legend : Actors and teams in Food-Microbiome projects. Round nodes depicts bioinformaticians or statisticians as follow: CP for Charlie PAUVERT, ALA for Anne-Laure ABRAHAM (PhD, Research Engineer), TG for Thibaut GUIRIMAND (Engineer), MM for Mahendra MARIADASSOU (PhD, Research Associate), SS for Sophie SCHBATH (PhD, Research Director), VL for Valentin LOUX (Engineer), SD for Sandra DEROZIER (Engineer). Square nodes depicts microbiologists as follow: PR for Pierre RENAULT (PhD, Research Director), BD for Bedis DRIDI (Post-doc).

```{r actors, engine="tikz",engine.opts=list(template="template-tikz2pdf.tex"),echo=FALSE, fig.cap="Actors and teams in Food-Microbiome projects. Round nodes depicts bioinformaticians or statisticians whereas square nodes depicts microbiologists. Underline names are my supervisors."}
\begin{tikzpicture}[
	every edge/.style = {draw=black,->, >=open triangle 60},
	bio/.style = {regular polygon,regular polygon sides=4,text=black,align=center,inner sep=0em},
	FM/.style = {draw=Igreen,rectangle,text=black,align=center},
	FMT/.style = {draw=Iorange,rectangle,text=black,align=center}]
\matrix [matrix of nodes, row sep = 1em,nodes={align=center,circle,draw},
		ampersand replacement=\&, nodes in empty cells,
		column sep={15em},
		column 1/.style ={text width = 2em},
		column 2/.style ={nodes={rectangle,draw=none},text width = 8em,inner sep=1em},
		column 3/.style ={text width = 2em} ] (m) {%
%\textbf{CP} \&[10em]  \&[10em] MM\\
\textbf{CP} \&  \& MM\\
\underline{ALA}\&       \& SS\\[1em]
|[bio]|\underline{PR} \&       \& |[draw=none,rectangle]|  \\[1em]
|[bio]|BD \&       \& VL\\
TG \&       \& SD\\
};
\node [draw,rectangle,dashed, fit={(m-1-1)(m-5-1)}, inner sep=4pt,label={[text width=8em,align=center]Unit{:} MICALIS\\Team{:} BAC}] (bac) {};
\node [draw,rectangle,dashed, fit={(m-1-3)(m-2-3)}, inner sep=4pt,label={[text width=10em,align=center]Unit{:} MaIAGE\\Team{:} StatInfOmics}] (mod) {};
\node [draw,rectangle,dashed, fit={(m-4-3)(m-5-3)}, inner sep=4pt,label={[text width=8em,align=center]Unit{:} MaIAGE\\Team{:} Migale}] (mig) {};
%\draw[<->, >=open triangle 60] (m-1-1) to [bend right=90] (m-2-1);
%\draw[<->, >=open triangle 60] (m-1-1) to [bend right=90] node [below,midway,sloped] {Supervisors} (m-3-1);
% Labels
\node [FM,fit={(m-1-2)(m-2-2)}, label={Food-Microbiome project}] (fm) {Shotgun metagenomics analysis tool};
\node [FM,fit={(m-4-2)(m-5-2)}, label={below:FM Transfert project}] (fmt) {Metadata-extended genomics database};
% FM-FMT relations
\draw [draw=black,->, >=open triangle 60] (fm) to [bend left] node[right] {Results} (fmt);
\draw [draw=black,->, >=open triangle 60] (fmt) to [bend left] node[left,near start] {Genomes} node [left,near end] {Annotations} (fm);
\draw (mod) edge node[above] {Statistical \& modelling} node[below] {expertise} (mod-|fm.east);
\draw (mig) edge node[above] {Platform administrators} node[below] {Technical advices} (mig-|fmt.east);
\draw (m-1-1) edge node[above] {Improvements \& tests}(m-1-1-|fm.west);
\draw (m-2-1) edge node[above] (id) {Initial developments} (m-2-1-|fm.west);
\draw[draw=black,->, >=open triangle 60] (m-3-1) to [bend left=10]  (fm.205);
\draw[draw=black,->, >=open triangle 60] (m-3-1) to [bend right=10] (fmt.150);
\node at (m-3-1-|id) {Biological expertise};
\draw (m-4-1) edge node[below] {Strains characterization} (m-4-1-|fmt.west);
\draw (m-5-1) edge node[below] {Developments \& design} (m-5-1-|fmt.west);
\end{tikzpicture}
```


## Cheese ecosystems particularities (1p)

* Define micro-organisms and taxonomy hierarchy.
* Ecosystems described as a _good_ model : both simple and complex (see Dutton)


### Why simple ?

* Numbers of species compared to others.
* Milk is an accessible media for culturing.
* Wide overall knowledge : literature available

### Why complex ?

* Sometimes (everytime) cheese is _not only_ milk and micro-organisms
* Environment influence is significant (others sources micro-organisms, possible interactions, abiotic parameters) (see Irlinger)

## State-of-art tools for ecosystem exploration (2.5-3p)

### Methods (1p)

* Focus on metagenomics.
* Different sequencing approaches and their limits.
* See Ercolini 2013 for next-generation sequencing and food microbial ecology

### Post sequencing strategies (1.5-2p)

* Presentation of different types of metagenomics data handling strategies.
* Different tools and their limits in ecosystem exploration.

## Aims

* Summary of a two-year apprenticeship.
* Exploration of strategy taking into account potential divergence between environmental and reference genomes.
