---
title: "Introduction"
author: Charlie PAUVERT
classoption: a4paper
fontsize: 12pt
lang: en
output:
  pdf_document:
    pandoc_args: [ "--latex-engine=/usr/bin/pdflatex" ] 
    toc: yes
    number_section: true
header-includes:
 - \usepackage{libertine}
--- 

# Introduction (7-15p)


## Food-Microbiome projects

We spend a fair amount of time eating, or thinking about it. 
Daily subjects are naturally studied and food -- as a complex product -- is investigated at several scales.
Multiple factors drives food-related studies: from scientific curiosity to yield and quality improvements.
Fermented foods - like cheese, sausage and beer - provides a level of diversity and complexity worth studying given their world-wide spread.

### Why? (1p)

Industrial partners and academics both wants insights into cheese ecosystems.
Their common interests were crystallized into two projects: (1) Food-Microbiome project and (2) Food-Microbiome Transfert project


####Food-Microbiome project

The Food-Microbiome project aims to deeply understand cheese ecosystems given next-generation sequencing technologies potential.
It is coordinated by Pierre RENAULT (PhD,Research Director) and funded by the ANR -_The French National Research Agency_-. It intends to provide a proof-of-concept and preliminary results concerning the use of metagenomics for cheese ecosystems exploration.

####Food-Microbiome Transfert project

The second project is divided into two parts: (1) the distribution of a metagenomics analysis tool and (2) the characterization of food-related strains.
<!-- The knowledge acquired during the previous project will be crystallized into a comprehensive metagenomics tool. -->
Metagenomics data analysis will be facilitated through an online interface where public and user-provided genomes could be used.
A comprehensive database and a website is developed to this end and target industrial partners and scientific collaborators.

Wet-lab experiments are planned by a post-doctoral researcher (Bedis DRIDI, PhD) in order to unravel technological abilities of unknown strains and improve genomics annotations.


### Who? (0.75-1p)

Farming provide most of our food, hence food-related studies are often led by INRA - _The French National Institute for Agricultural Research_.

INRA is an EPST - _Scientific and Technological Public Structure_ - which yields scientific knowledge concerning food, agriculture and environment since 1946.
It is organized into well-defined scientific divisions or _department_ -e.g., MICA for _Microbiology and the Food Chain_- which gather several research _units_ -e.g., MICALIS for _Food microbiology for health_-.
INRA is spread across the country in _centers_ and scientists work locally in _teams_ that are part of units.

#### Several actors and contributors to these projects

Food-Microbiome projects are led by teams in a major INRA center: Jouy-en-Josas (FR).
They are backed by teams in others centers such as Grignon (FR) and Aurillac (FR), respectively led by F.IRLINGER and M-C. MONTEL.

Science implies collaborations and these projects includes industrials partners gathered in CNIEL -_The French Dairy Interbranch Organization_.

**Food and Commensal Bacteria Team** BAC Team is led by Pierre RENAULT (PhD, Research Director, abbrev: PR).
Anne-Laure ABRAHAM (PhD, Research Engineer, abbrev: ALA), Thibaut GUIRIMAND (Engineer, abbrev: TG), and myself Charlie PAUVERT (abbrev: CP) constitute the development bioinformatics team. Bedis DRIDI (Post-doc, abbrev: BD) is implied in wet lab experiments.

* MaIage unit :
	+ VL, SD, (TG): infrastructure team, platform administrator, technical advices
	+ MM, SB: statistical expertise.


#### Personal involvement in both projects

* Mainly development and tests in FM.
* Knowledge/know-how transfer and integration in FMT.

Legend of Figure \ref{fig:actors} CP for Charlie PAUVERT, ALA for  MM for Mahendra MARIADASSOU (PhD, Research Associate), SS for Sophie SCHBATH (PhD, Research Director), VL for Valentin LOUX (Engineer), SD for Sandra DEROZIER (Engineer). 


```{r actors, engine="tikz",engine.opts=list(template="template-tikz2pdf.tex"),echo=FALSE, fig.align="center",fig.scap="Actors and teams in Food-Microbiome projects.", fig.cap="Actors and teams in Food-Microbiome projects. Round nodes depicts bioinformaticians or statisticians whereas square nodes depicts microbiologists. Underline names are my supervisors. Initials are detailed in the text.",cache=TRUE}
\input{figures/fig-actors}
```


## Cheese ecosystems particularities (1p)

###Micro-organisms and definition of species

Micro-organisms unite bacteria, yeasts, molds and viruses.
These micro-organisms are usually classified using taxonomic ranks as illustrated in \ref{fig:taxonomic}.

```{r taxonomic, engine="tikz",engine.opts=list(template="template-tikz2pdf.tex"),echo=FALSE,fig.align="center", fig.pos="h",fig.cap="Illustration of taxonomic ranks. Current limits of taxonomical identification in several metagenomics sequencing strategies are described by symbols: diamonds ($\\diamond$) depicts shotgun sequencing technology whereas circles ($\\circ$) depicts amplicon sequencing strategies."}
\input{figures/fig-taxonomic}
```

However whole-genome sequencing blurred these ranks and were thought to be augmented by genomics information [@chun_integrating_2014].
DNA-DNA hybridization and its _in silico_ counterpart are proposed to delineate species limits [@richter_shifting_2009; @varghese_microbial_2015].
For convenience I will use the standard classification when referring to micro-organisms.

### Cheese ecosystems

Cheese-making starts with a matrix composed of milk more or less transformed.
Proteins, fatty acids and water -- composing the milk -- combined with tailored conditions leads to an efficient microbial growth [@monnet_growth_2015].


These ecosystems are described as a _good_ model: both simple and complex [@wolfe_cheese_2014].


###Why simple?


* Numbers of species compared to others.
* Milk is an accessible media for culturing.
* Wide overall knowledge : literature available

NGS technologies shed light on a microbial diversity in what was thought to be well-described ecosystems [@quigley_high-throughput_2012].

###Why complex ?

* Sometimes (everytime) cheese is _not only_ milk and micro-organisms
* Environment influence is significant (others sources micro-organisms, possible interactions, abiotic parameters) (see Irlinger)

Despite rich media and supposed easy culture, low abundant micro-organisms in cheese ecosystems remains uncharacterized.

## State-of-art tools for ecosystem exploration (2.5-3p)

### Methods (1p)

* Focus on metagenomics.
* Different sequencing approaches and their limits.
* See Ercolini 2013 for next-generation sequencing and food microbial ecology

```{r ecosys, engine="tikz",engine.opts=list(template="template-tikz2pdf.tex"),echo=FALSE, fig.align="center",fig.cap="Overview of microbial communities exploration using metagenomics methods. Bold items describe technologies and methods used in our projects."}
\input{figures/fig-ecosys}
```

### Post sequencing strategies (1.5-2p)

* Presentation of different types of metagenomics data handling strategies.
* Different tools and their limits in ecosystem exploration.

```{r methods, engine="tikz",engine.opts=list(template="template-tikz2pdf.tex"),echo=FALSE, fig.align="center",fig.cap="Comparison of post-sequencing strategies."}
\input{figures/fig-methods-cmp}
```

## Aims

* Summary of a two-year apprenticeship.
* Exploration of strategy taking into account potential divergence between environmental and reference genomes.
