---
title: "Materials and Methods"
author: Charlie PAUVERT
classoption: a4paper
fontsize: 12pt
lang: en
output:
  pdf_document:
    pandoc_args: [ "--latex-engine=/usr/bin/pdflatex" ] 
    toc: yes
    number_section: true
header-includes:
 - \usepackage{libertine}
--- 



#Materials and methods (7p)

##Computing facilities

**Personal computers**
The institute computer I am working on runs under Microsoft Windows 7 64bits.
Its features --Intel Xeon 2.66Ghz and 8Go RAM-- support a virtual machine set up.
Through VirtualBox I can work under a suitable operating system for my needs: Linux Debian “wheezy“ 7.8 (stable).
From times to times, I use my personal laptop for writings during commutes or remote working.

**Clusters farms**
Some greedy simulations or tasks requires supplementary computational resources.
I have been granted access to two INRA clusters farms to meet these needs.
The first one --`migale`-- located in Jouy-en-Josas harbours 580 nodes and a large choice of bioinformatics tools.
The second one --`genotoul`-- in Toulouse benefits from a larger storage space and has more than 5,000 nodes.


##Professional practice

###Technology and literature monitoring

#####Literature monitoring
I usually browse state-of-the art journals --e.g., *Bioinformatics*, *Plos Computational Biology*-- outlines in order to quickly get a glance at new articles.
Word of mouth and informal discussions are far from outdated and relevant to pin point interesting papers.
This network is defined at several scales:
(i) at office level with Anne-Laure et Pierre,
(ii) with team members,
(iii) through collaborations and others informals interactions within the institute --e.g., young researchers association, bioinformaticians network-- and finally
(iv) within master degrees colleagues. 


We set up and promote a wiki --within the institute-- dedicated to biologists and bioinformaticians from the MICALIS research unit.
Articles and tips are shared through this platform with colleagues working on similar topics.
A shared bibliography was set up especially with MetaGenoPolis [^1].

My bibliography is managed with [Zotero (v4.0)](https://www.zotero.org/). This tool provide remote synchronisation between the several terminals I use --lab computer and personal computer.
Articles of interests are hence gathered both at the lab or at the university.

To circumvent Bib\LaTeX-related issues --such as non-unique citations keys or specials characters-- I have been using [Better Bib\LaTeX](https://github.com/retorquere/zotero-better-bibtex) Zotero extension which tackle previous issues and provide automatic bibliography exports easily.

#####Technology watch
Many resources exists to keep up with bioinformatics and biostatistics related activities.
Mailing lists from SFBI --_French Bioinformatics Society_-- or AFEM --_French Microbial Ecology Association_-- or more specific to the institute are examples of resources.
I usually browse SFBI "bioinformations", [bioinfo-fr.net website](bioinfo-fr.net), or [R-bloggers](http://www.r-bloggers.com/) for general information and [stackoverflow.com forum](stackoverflow.com) --and its sub-components-- to answer technical issues.


#####Meetings, work groups and seminars
Science without regular communication with peers would not be.
It is one way to learn new methods and share feedbacks.
Adapting your speech to a public with different scientifc background is a good opportunity to synthetise your research topic. 
To this end, I had attended monthly bioinformatics meetings organized in the institute.

I was also involved in an INRA working group belonging to the PEPI network --_Experience and good practices sharing in Computer Science_.
Bioinformaticians and statisticians gathered in this group are interested by amplicon metagenomics analysis --especially 16S rDNA.
I could compare these approaches and biases outlined with my project. 
I participated to the beta test of the training course organized by this working group.
These meetings provide a new perspective to my project and in the long term to metagenomics analysis.

Conferences are a common way of keeping yourself up to date in a field as broad as bioinformatics.
I attended three national and international conferences such as JOBIM --_Open Days in Biology, Computer Science and Mathematics_-- or ECCB --_European Conference on Computational Biology_ throughout my apprenticeship.
I also attended a yearly workshop RCAM --_Recent Computational Advances in Metagenomics_-- where I could discuss with experts of the field.

###Good practices and tracability

##### vim

##### git


I created a private [GitHub](http://github.com/cpauvert) repository to manage versions of this thesis. It also provides online storage and availability from anywhere.

#####GeDI development

###Research communication

In February 2016, I presented my work on the mixture model to the monthly bioinformatics meeting of the institute despite negative results.


##Tools

#####Short reads alignment on reference genome

`bowtie` version 1 because cheese samples were sequenced with SOLiD technology.

#####Synthetic metagenomics samples creation

#####In between genomes identity percentage computation

#####Bioinformatics toolbox

`SAMtools` (version) were used in the last version of our tool.
I relied on `BEDtools` (version) to provide fast and efficient alignment file and gene annotation intersection.
These toolbox were either called from our tool with a Python library `pysam` (v0.14.1) or through the Python `os` module.

#####Data mining and explorative analyses

####Metagenomics analysis tools

#####GeDI last year

```{r former-gedi,  engine="tikz",engine.opts=list(template="template-tikz2pdf.tex"),echo=FALSE, fig.align="center",fig.scap="Previous data flow: GeDI", fig.cap="Previous data flow: GeDI",cache=FALSE }
\input{figures/fig-former-gedi}
```

#####GeDI standard files

```{r gedi,  engine="tikz",engine.opts=list(template="template-tikz2pdf.tex"),echo=FALSE, fig.align="center",fig.scap="Overview of our metagenomics analysis tool: GeDI", fig.cap="Overview of our metagenomics analysis tool: GeDI. It consists of three Python (v2.7.5) modules interconnected. Each module is framed here with dashed lines and encompass data and software.",cache=FALSE }
\input{figures/fig-gedi}
```

###Mixture model of distributions {#modele}

##### Contribution

##### Distribution

##### Modele

##### Parameters estimation

#####Expectation-Maximization algorithm

##Data

#####Micro-organisms genomes used

#####_Streptococcus_ training datasets
I simulated a shotgun sequencing for each _Streptococcus_ strains listed in the table \ref{tab:strep}.
These reads are then aligned separetely on _one_ reference genome: the strain _Streptococcus salivarius_ JIM8777 using the former version of GeDI--see \ref{fig:gedi}


  **Strains**                                 **ANI** (%)     **Distance**       **Proximity**
  ------------------------------------------ ---------------- ----------------- ---------------
  _Streptococcus salivarius_ JIM8777              100.0       Exact strain             0
  _Streptococcus salivarius_ NCTC 8618            89.63       Close strain             1
  _Streptococcus salivarius_ CCHSS3               88.57       Close strain             1
  _Streptococcus salivarius_ K12                  86.01       Close strain             1
  _Streptococcus vestibularis_ ATCC 49124         80.01       Sub-species              2
  _Streptococcus thermophilus_ JIM 8232           74.68       Sub-species              2
  _Streptococcus salivarius_ PS4                  73.74       Sub-species              2
  _Streptococcus agalactiae_ NEM316               67.14       Close species            3
  _Streptococcus infantarius_ ATCC BAA-102        62.82       Close species            3
  _Streptococcus mutans_ UA159                    59.86       Distant species          3

Table: _Streptococcus_ dataset composition and distance to the reference strain (_Streptococcus salivarius_ JIM8777). Closeness classes are based on ANI _Average Nucleotide Identity_ computed with Gegenees [@agren_gegenees:_2012].

```{r strep,  engine="tikz",engine.opts=list(template="template-tikz2pdf.tex"),echo=FALSE, fig.align="center",fig.scap="Training dataset overview", fig.cap="Training dataset overview.",cache=TRUE }
\input{figures/fig-strep}
```


#####Annotation files
Standard files for gene annotation --GFF3-- were fetched from either from public databases like NCBI or from within the institute databases. 
In the absence of such file type, I sometimes used a BioPerl script to convert GenBank files to a pair of FASTA file and GFF file.

#####Mock Community


* **Insist on EM algorithm** according to NV remarks?
* New python libraries used in GeDI.


```{r coverage,  engine="tikz",engine.opts=list(template="template-tikz2pdf.tex"),echo=FALSE, fig.align="center",fig.scap="Genome coveragecomputation after reads alignement.", fig.cap="Genome coverage computation after reads alignement. In our metagenomics analysis tool --GeDI-- we discrimate (1) coverage ($c_{\\text{Total}}$) yield by the number of reference genome positions where at least one read is mapped divided by genome length ($l_G$) and (2) coverage ($c_{\\text{Mismatchs}}$) concerning the same metric only for reads aligned with one to three mismatches compared to the reference.",cache=FALSE }
\input{figures/fig-coverage}
```

[^1]: Industrial and academics consortium tied with INRA working on human gut microbiota using metagenomics.
