---
title: "Materials and Methods"
author: Charlie PAUVERT
classoption: a4paper
fontsize: 12pt
lang: en
output:
  pdf_document:
    pandoc_args: [ "--latex-engine=/usr/bin/pdflatex" ] 
    toc: yes
    number_section: true
header-includes:
 - \usepackage{libertine}
--- 



#Materials and methods (7p)

##Computing facilities

**Personal computers**
The institute computer I am working on runs under Microsoft Windows 7 64bits.
Its features --Intel Xeon 2.66Ghz and 8Go RAM-- support a virtual machine set up.
Through VirtualBox I can work under a suitable operating system for my needs: Linux Debian “wheezy“ 7.8 (stable).
From times to times, I use my personal laptop for writings during commutes or remote working.

**Clusters farms**
Some greedy simulations or tasks requires supplementary computational resources.
I have been granted access to two INRA clusters farms to meet these needs.
The first one --`migale`-- located in Jouy-en-Josas harbours 580 nodes and a large choice of bioinformatics tools.
The second one --`genotoul`-- in Toulouse benefits from a larger storage space and has more than 5,000 nodes.


##Professional practice

###Technology and literature monitoring

#####Literature monitoring
I usually browse state-of-the art journals --e.g., *Bioinformatics*, *Plos Computational Biology*-- outlines in order to quickly get a glance at new articles.
Word of mouth and informal discussions are far from outdated and relevant to pin point interesting papers.
This network is defined at several scales:
(i) at office level with Anne-Laure et Pierre,
(ii) with team members,
(iii) through collaborations and others informals interactions within the institute --e.g., young researchers association, bioinformaticians network-- and finally
(iv) within master degrees colleagues. 


We set up and promote a wiki --within the institute-- dedicated to biologists and bioinformaticians from the MICALIS research unit.
Articles and tips are shared through this platform with colleagues working on similar topics.
A shared bibliography was set up especially with MetaGenoPolis [^1].

My bibliography is managed with [Zotero (v4.0)](https://www.zotero.org/). This tool provide remote synchronisation between the several terminals I use --lab computer and personal computer.
Articles of interests are hence gathered both at the lab or at the university.

To circumvent Bib\LaTeX-related issues --such as non-unique citations keys or specials characters-- I have been using [Better Bib\LaTeX](https://github.com/retorquere/zotero-better-bibtex) Zotero extension which tackle previous issues and provide automatic bibliography exports easily.

#####Technology watch
Many resources exists to keep up with bioinformatics and biostatistics related activities.
Mailing lists from SFBI --_French Bioinformatics Society_-- or AFEM --_French Microbial Ecology Association_-- or more specific to the institute are examples of resources.
I usually browse SFBI "bioinformations", [bioinfo-fr.net website](bioinfo-fr.net), or [R-bloggers](http://www.r-bloggers.com/) for general information and [stackoverflow.com forum](stackoverflow.com) --and its sub-components-- to answer technical issues.


#####Meetings, work groups and seminars
Science without regular communication with peers would not be.
It is one way to learn new methods and share feedbacks.
Adapting your speech to a public with different scientifc background is a good opportunity to synthetise your research topic. 
To this end, I attend monthly bioinformatics meetings organized in the institute.
In February 2016, I presented my work on the mixture model to this group despite negative results.

I was also involved in an INRA working group belonging to the PEPI network --_Experience and good practices sharing in Computer Science_.
Bioinformaticians and statisticians gathered in this group are interested by amplicon metagenomics analysis --especially 16S rDNA.
I could compare these approaches and biases outlined with my project. 
I participated to the beta test of the training course organized by this working group.
These meetings provide a new perspective to my project and in the long term to metagenomics analysis.

Conferences are a common way of keeping yourself up to date in a field as broad as bioinformatics.
I attended three national and international conferences such as JOBIM --_Open Days in Biology, Computer Science and Mathematics_-- or ECCB --_European Conference on Computational Biology_ throughout my apprenticeship.

* How do I work?
* What tools and data did I used?
* **Insist on EM algorithm** according to NV remarks?
* New python libraries used in GeDI.
* Private GitHub for report.

```{r gedi,  engine="tikz",engine.opts=list(template="template-tikz2pdf.tex"),echo=FALSE, fig.align="center",fig.scap="Overview of our metagenomics analysis tool: GeDI", fig.cap="Overview of our metagenomics analysis tool: GeDI. It consists of three Python (v2.7.5) modules interconnected. Each module is framed here with dashed lines and encompass data and software.",cache=FALSE }
\input{figures/fig-gedi}
```

```{r coverage,  engine="tikz",engine.opts=list(template="template-tikz2pdf.tex"),echo=FALSE, fig.align="center",fig.scap="Genome coveragecomputation after reads alignement.", fig.cap="Genome coverage computation after reads alignement. In our metagenomics analysis tool --GeDI-- we discrimate (1) coverage ($c_{\\text{Total}}$) yield by the number of reference genome positions where at least one read is mapped divided by genome length ($l_G$) and (2) coverage ($c_{\\text{Mismatchs}}$) concerning the same metric only for reads aligned with one to three mismatches compared to the reference.",cache=FALSE }
\input{figures/fig-coverage}
```

[^1]: Industrial and academics consortium tied with INRA working on human gut microbiota using metagenomics.
