# Outline : around a metagenomics analysis tool

Ecosystems particularities leads to :

1. identify  limits in existing **metagenomics tools**
2. develop or adapt aforementioned tools or create a new one.


A two-year work around a metagenomics analysis tool `GeDI` :

## Improvements

- in terms of code (file size, speed, etc.) cf. Integration
- or exploration of taxonomic classification :
	+ Mixture model proof-of-concept
	+ Model limits and (partial) solutions

## Comparison

- to existing tools (at the time)
- to new tools (`Sigma`, `MicrobeGPS`)

## Integration

- with FoodMicrobiome Transfert
- with Galaxy Platform
- with others tools through the use of bioinformatics standard files.

## Application

- to simulated datasets
- to cheese metagenomics analysis : 6 samples


# Changes to introduce

## Introduction


### Work context

* **Project context**: Insist on FM and FM-T project.
* Replace every role, especially mine: interfacing both projects.
* Figure from CNIEL presentation.
* Mention TG +BD in team member/collaborations: bioinformatics and microbiology interface.
* Collaborations with SD and VL. (Mention Galaxy and OR here or later? )

## Materials and methods

* Insist on EM algorithm according to NV remarks?
* New python librairies used in GeDI.
* Private GitHub for report.


## Results

### Modeling exploration results

* Limits identification and solutions hunt.
* Show posterior probabilities expected usage despite few results?

### Metagenomics analysis results

* Cheese analysis results
* Message: are mismatches information an addition to metagenomics analysis?


## Prospects

* Tool will be used by industrials and academics.
* Specific formation will be provided throught the Migale platform.
* Details on how to compare tools with others in case of time constraints.
