# Outline : around a metagenomics analysis tool

Ecosystems particularities leads to :

1) identify  limits in existing **metagenomics tools** 
2) develop or adapt  


Tool GeDI :

## Improvements

- in terms of code (file size, speed, etc.) cf. Integration
- or exploration of taxonomic classification :
	+ Mixture model proof-of-concept 
	+ Model limits and (partial) solutions

## Comparison

- to existing tools (at the time)
- to new tools (Sigma, MicrobeGPS)

## Integration

- with FoodMicrobiome Transfert
- with Galaxy Platform
- with others tools through the use of bioinformatics standard files.

## Application

- to simulated datasets
- to cheese metagenomics analysis : 6 samples


# Changes to introduce 

## Introduction

### Work context

* Add a paragraph for FoodMicrobiome-Transfert Project.
* Mention TG in team member/collaborations
* Collaborations with SD and VL. (Mention Galaxy and OR here or later? )

## Materials and methods

* Insist on EM algorithm according to NV remarks?
* New python librairies used in GeDI.
* Private GitHub for report.


