---
title: "Results"
author: Charlie PAUVERT
classoption: a4paper
fontsize: 12pt
lang: en
output:
  pdf_document:
    pandoc_args: [ "--latex-engine=/usr/bin/pdflatex" ] 
    toc: yes
    number_section: true
header-includes:
 - \usepackage{libertine}
--- 

# Results (22p)

In the scope of Food-Microbiome projects, we rely on existing reference genomes to (1) identify micro-organisms present in the ecosystem -- if possible to the *strain* level -- and (2) characterize low-abundant organisms.


I this chapter, I will focus on my two-year work around a metagenomics analysis tool `GeDI` through four main axes.
I will present (1) several approaches explored to improve the tool, followed by (2) a comparison with similar tools and (3) how this module was integrated to conclude with (4) results from its application on simulated and real datasets.


## Improvements

* exploration of taxonomic classification :
	+ Mixture model proof-of-concept
	+ Model limits and (partial) solutions
* in terms of code (file size, speed, etc.) cf. Integration

### Modeling approaches

* Previous approaches in the lab or during my internship

```{r previous-approaches, engine="tikz",engine.opts=list(template="template-tikz2pdf.tex"),echo=FALSE, fig.align="center",fig.scap="Previous modeling approaches summary embedded in GeDI.", fig.cap="Previous modeling approaches summary embedded in GeDI. Theses differents approaches were tested at different period depicted as symbols: before my internship ($\\circ$), during my first year of master degree internship ($\\diamond$) and during my two-year apprenticeship ($\\star$).",cache=TRUE}
\input{figures/fig-previous-approaches}
```

* Mixture model principle (details in Materials chapter)

### Modeling exploration results

* Limits identification and solutions hunt.
* Possible usage.

## Comparison

* to existing tools (at the time)
* to new tools (`Sigma`, `MicrobeGPS`)

This part could be in prospects if time is running out.

## Integration

* with FoodMicrobiome Transfert
* with Galaxy Platform
* with others tools through the use of bioinformatics standard files.

## Application

* to simulated datasets
* to cheese metagenomics analysis : 6 samples
