\pgfmathsetseed{3}
\begin{tikzpicture}
%\tikzstyle{bact}=[rectangle, minimum width=4em,minimum height=3ex,rounded corners=0.2ex]
\tikzstyle{bact}=[rectangle, minimum width=8em,minimum height=6ex,rounded corners=0.4ex]
\tikzstyle{lecture}=[rectangle, minimum width=3em,minimum height=1em]
\tikzstyle{marqueur}=[rectangle, minimum width=3em,minimum height=1em, draw=black, thick]

\matrix [matrix of nodes, nodes in empty cells,nodes={align=center},row sep = 2em,
		ampersand replacement=\&,
		column 2/.style ={text width = 11em},
		column sep={3em}] (m) {%
			% row 1
|[text width =2cm, text centered]|  {Microbial ecosystem} \&   |[rectangle, draw=Igreen, fill=Igreen!25,align=left]| {$\bullet$ Which micro-organisms \par ~~are present? \\ $\bullet$ What functions \par ~~are associated?} \& \\
			% row 2
			\&  |[rectangle, draw=Igreen, label={\textbf{Metagenomics}}]| {Capture and analysis \\of ecosystem \\ genomics information}\& \\[2em]
			% row 3
			|[fill=Iblue!25,text width=3.5cm,draw]|  {\textbf{Shotgun sequencing} (whole genome)} \&  \& |[fill=Iorange!25,draw, text width=3.5cm, text centered]| {Amplicon sequencing (marker genes,...)}\\
			% row 4
 \& {Millions of DNA reads} \&  \\
};

  \begin{scope}[x=15em, y=10em,scale=0.1, every node/.style={transform shape}, local bounding box=petri, shift={($(m-3-2)$)}]
  \foreach \n /\col in {4/Igreen, 12 /Iblue, 4 /Iblue!65, 6 /Iorange,4/Iblue!65!black,2/Igreen!65!black}{%
	  \foreach \x in {1,...,\n}{
	      \node[bact, draw=\col, rotate =180*rand] (\x) at (rand, rand) {};
	  }
  };
  \end{scope}  

  \begin{scope}[x=15em, y=7em, scale=0.2, every node/.style={transform shape}, local bounding box=seq, shift=(m-4-1)]
  \foreach \n /\col in {4/Igreen, 20 /Iblue, 10 /Iblue!65, 20 /Iorange,10/Iblue!65!black,8/Igreen!65!black}{%
	  \foreach \x in {1,...,\n}{
	      \node[lecture, fill=\col] (l\x) at (rand, rand) {};
	  }
  };
  \end{scope} 
  \begin{scope}[x=15em, y=7em, scale=0.2, every node/.style={transform shape}, local bounding box=cib, shift=(m-4-3)]
  \foreach \n /\col in {4/Igreen, 20 /Iblue, 10 /Iblue!65, 20 /Iorange,10/Iblue!65!black,8/Igreen!65!black}{%
	  \foreach \x in {1,...,\n}{
	      \node[marqueur, draw=\col, fill=black] (m\x) at (rand, rand) {};
	  }
  };
  \end{scope} 
  \node[below=of m-4-1,rectangle split horizontal, rectangle split, rectangle split parts=3 , text width=3cm, text centered, draw, fill=Iblue!25] (id) {\textbf{Existing species and strains identification}\nodepart{second} De-novo genome assemblies \nodepart{third}Metabolism prediction};
  \node[below=of m-4-3,rectangle split horizontal,rectangle split, rectangle split parts =2, text width=2.5cm, text centered, draw, fill=Iorange!25] (id16) {Existing genus and species identification\nodepart{two} Micro-organisms quantification};
  \node[circle, draw,fit= (petri),label={above:Sampling}] (fitP) {};
\draw[thick , ->, >=stealth] (m-1-1) -- (m-1-1-|m-1-2.west);
\draw[thick , ->, >=stealth] (m-1-1) |- (m-2-2);
\draw[thick , ->, >=stealth] (fitP) -- node[above,midway] {DNA} node[below,midway] {extraction} (fitP-|m-3-1.east);
\draw[thick , ->, >=stealth] (fitP) -- node[above,midway] {DNA} node[below,midway] {extraction} (fitP-|m-3-3.west);
%
\node[below=of id.west,yshift=-1cm,xshift=1cm, anchor=north west,text width=3.8cm, text centered,dash pattern = on 2pt off 3pt on 4pt off 4pt, draw] (cmp){Reads composition comparison to known profiles};
\node[below=1em of cmp,text width=4cm, text centered,dash pattern = on 2pt off 3pt on 4pt off 4pt, draw](alg) {\textbf{Reads mapping to reference genomes}};
\node[below=of id16.west,yshift=-1cm,xshift=1cm, anchor=north west, text width=4cm,text centered,dash pattern = on 2pt off 3pt on 4pt off 4pt, draw](alg16) {Reads comparison to a marker genes bank };
%
\draw[->,>=stealth] (seq) -- (id) ;
\draw[->,>=stealth] (seq) -- ($(id.north east)+(-1.25cm,0)$) ;
\draw[->,>=stealth] (seq) -- ($(id.north west)+(1.25cm,0)$) ;
\draw[->,>=stealth] (cib) -- ($(id16.north)+(-1.25cm,0)$) ;
\draw[->,>=stealth] (cib) -- ($(id16.north)+(1.25cm,0)$) ;
\draw[->,>=stealth] (id.south west) ++(0.5cm,0) |- (cmp.west) ;
\draw[->,>=stealth] (id.south west) ++(0.5cm,0) |- (alg.west) ;
\draw[->,>=stealth] (id16.south west) ++(0.5cm,0) |- (alg16.west) ;
\node[draw=lightgray, inner sep=3pt,fit={(id) (id16)},label={below:Post-sequencing \textbf{methods}}]  {};
\node[draw=lightgray, inner sep=10pt,fit={(cmp) (alg) (alg16)},label={below:Micro-organisms identification \textbf{strategies} examples}] {};
\end{tikzpicture}
