\begin{tikzpicture}[
	soft/.style = {draw=Igreen,signal,signal to=east, fill=Igreen!25,text=black,text width=6em},
	data/.style = {draw=Igreen,tape,tape bend top=none,fill=white,text=black,align=center,double copy shadow={opacity=.5},text width=6em},
	alert/.style = {very thick},
	every pin/.style = {font=\footnotesize},
	link/.style = {draw=black,->, >=stealth}]
\matrix [matrix of nodes, row sep = 1em,column sep=6em,nodes={align=center},ampersand replacement=\&
		%column 1/.style ={text width = 6em},
		%column 2/.style ={text width = 6em},
		%column 3/.style ={text width = 8em}
] (m) {%
	|[data,pin={right:FASTA}]| Reference Genomes\&[-4em] \&  |[data,pin={[name=fq]right:FASTQ}]| Sequencing Data\\
	|[soft,label={below:\texttt{bowtie-build}}]| Reference creation\& |[data,pin={below:BWT}]| Genomes Indexes \&  |[soft,text width=8em,label={below:\texttt{bowtie v1}}]| Alignment on reference genomes\\
	\& |[data,pin={[name=gff]right:GFF}]| Genes Annotations\&  |[data,text width=8em,pin={[name=bam]right:BAM}]| Alignment file for each genome\\
 \& \&  |[soft]| BAM--GFF Intersection \\
	\& \& |[data,pin={[name=out]right:CSV $+$ GFF}]| Output files \\
			%\& \& |[data,visible on=<3->]| Composition de l'écosystème et Détails par génome \\
};

\matrix [row sep=.1em, nodes={rectangle, minimum width=4em, minimum height=.01em, fill}, ampersand replacement=\&] (genome) at (m-1-1-|m-2-2) {%
	\node [Iblue!85!black] {};\& \\
	\node [Iblue!85] {};\& \\
	\node [Iblue!65] {};\& \\
	\node [Iorange!85!black] {};\& \\
	\node [Ipurple] {};\& \\};
\tikzstyle{lecture}=[rectangle, minimum width=3em,minimum height=1em]
\pgfmathsetseed{3}
\node[left=2em of m-1-3] (nL) {};
\begin{scope}[x=15em, y=7em, scale=0.1, every node/.style={transform shape}, local bounding box=seq, shift={($(nL)$)}]
  \foreach \n /\col in {4/Igreen, 20 /Iblue, 10 /Iblue!65, 20 /Iorange,10/Iblue!65!black,8/Igreen!65!black}{%
	  \foreach \x in {1,...,\n}{
	      \node[lecture, fill=\col] (l\x) at (rand, rand) {};
	  }
  };
  \end{scope} 
  \draw[link] (m-1-1) edge (m-2-1);
\draw[link] (m-2-1) edge (m-2-2);
\draw[link] (m-2-2) edge (m-2-3);
\draw[link] (m-1-3) edge (m-2-3);
\draw[link] (m-2-3) edge (m-3-3);
\draw[link] (m-3-3) edge (m-4-3);
\draw[link] (m-2-1) |- (m-3-2);
\draw[link] (m-3-2) |- node[above,near end] {CDS Filter} (m-4-3);
\draw[link] (m-4-3) edge (m-5-3);
%
\node [thick,draw=Iblue,dashed, fit={(m-1-1)(m-2-1)(m-3-2)(genome)(gff)},inner sep=10pt,label={(1) GeDI Reference Creator}] {};
\node [thick,draw=Iorange,dashed, fit={(m-1-3)(m-2-3)(m-3-3)(seq)(fq)(bam)}, inner sep=10pt,label={(2) GeDI Complete Genome}] {};
\node [thick,draw=Ipurple,dashed,fit={(m-3-2)(m-3-3)(m-5-3)(m-4-3)(out)(bam)},inner sep=10pt,label={below:(3) GeDI Summary by Genes}] {};
\end{tikzpicture}
